static/js/message_events.js

```diff
@@ -259,6 +260,15 @@ export function update_messages(events) {
                 compose_fade.set_focused_recipient("stream");
             }
 
+            const current_drafts = drafts.draft_model.get();
+            for (const draft_id of Object.keys(current_drafts)) {
+                const draft = current_drafts[draft_id];
+                if (draft.topic === orig_topic) {
+                    draft.topic = new_topic;
+                    drafts.draft_model.editDraft(draft_id, draft, false);
+                }
+            }
```

Comment:
From an abstractions standpoint, I think we want this to be a function exported by `drafts.js`